src/state/routing/quickRouteSlice.ts

```diff
@@ -0,0 +1,112 @@
+import { createApi, fetchBaseQuery, FetchBaseQueryError } from '@reduxjs/toolkit/query/react'
+import { sendAnalyticsEvent } from 'analytics'
+import ms from 'ms'
+import { logSwapQuoteRequest } from 'tracing/swapFlowLoggers'
+import { trace } from 'tracing/trace'
+
+import { GetQuickQuoteArgs, PreviewTradeResult, QuickRouteResponse, QuoteState, RouterPreference } from './types'
+import { isExactInput, transformQuickRouteToTrade } from './utils'
+
+const UNISWAP_API_URL = process.env.REACT_APP_UNISWAP_API_URL
+if (UNISWAP_API_URL === undefined) {
+  throw new Error(`UNISWAP_API_URL must be a defined environment variable`)
+}
+
+function getQuoteLatencyMeasure(mark: PerformanceMark): PerformanceMeasure {
+  performance.mark('quickroute-fetch-end')
+  return performance.measure('quickroute-fetch-latency', mark.name, 'quickroute-fetch-end')
+}
+
+export const quickRouteApi = createApi({
```

Comment:
curious why you made this its own api as opposed to adding an endpoint to the existing routing api